src/cli/self_update/windows.rs

```diff
@@ -500,13 +500,23 @@ pub(crate) fn do_remove_from_path() -> Result<()> {
 
 const RUSTUP_UNINSTALL_ENTRY: &str = r"Software\Microsoft\Windows\CurrentVersion\Uninstall\Rustup";
 
+fn rustup_uninstall_reg_key() -> Result<RegKey> {
+    Ok(RegKey::predef(HKEY_CURRENT_USER)
+        .create_subkey(RUSTUP_UNINSTALL_ENTRY)
+        .context("Failed creating uninstall key")?
+        .0)
+}
+
+pub(crate) fn do_update_programs_display_version(version: &str) -> Result<()> {
+    rustup_uninstall_reg_key()?
+        .set_value("DisplayVersion", &version)
+        .context("Failed to set display version")
```

Comment:
perhaps either DisplayVersion (as it is a concrete thing) or

`Failed to record version for uninstall` (as this is the conceptual layer)

Reply:
@rbtcollins That's the original error message. But I agree we can change it.

Reply:
Done.